storage/paths/local.go

```diff
@@ -438,7 +468,24 @@ func (st *Local) Reserve(ctx context.Context, sid storiface.SectorRef, ft storif
 
 	st.localLk.Lock()
 
-	done := func() {}
+	var firstDonebuf []byte
+	var firstDoneN int
+
+	var doneCalled bool
+	done := func() {
+		if doneCalled {
+			curStack := make([]byte, 20480)
+			curStackN := runtime.Stack(curStack, false)
+
+			log.Errorw("double done call", "sector", sid, "fileType", ft, "prevStack", string(firstDonebuf[:firstDoneN]), "curStack", string(curStack[:curStackN]))
+		}
+
+		firstDonebuf = make([]byte, 20480)
+		firstDoneN = runtime.Stack(firstDonebuf, false)
+
+		doneCalled = true
+	}
+
 	deferredDone := func() { done() }
 	defer func() {
 		st.localLk.Unlock()
```

Comment:
Lets get rid of the confusing done(): 
1. Name Reserve()'s returns: (release func(), err error)
2. Here, if err != nil { log_the_stack } // is this the intent of the first done() ? 
3. The L527 done() could be inline. 
A double-done should be impossible.

Reply:
> Here, if err != nil { log_the_stack } // is this the intent of the first done() ?

No, the intent of this stack logging was identifying where this double call was happening.

Can drop that code, but decided to leave as it doesn't do much harm, and if we ever hit that again the visibility will be very handy.

Reply:
> The L527 done() could be inline.

Can you elaborate what do you mean by that? That code just appends some cleanup logic to the `done` call chain; The call chain will only ever be called once by the defer if Reserve fails at any later point, or if it succeeds, by the release return.

The bug here was that in the successful case the release function could be called more than once by the task_storage code, which was a bug in that code.